### ISSUE TITLE
ERROR: "Received binary image input but resource not configured to accept NHWC layout"

### ISSUE THREAD

**Issue by owlet42** [openvinotoolkit/model_server#912]
I followed this manual(https://github.com/openvinotoolkit/model_server/blob/develop/docs/ovms_onnx_example.md#option-2-adding-preprocessing-to-the-server-side-building-a-dag-) to experiment and encountered an error

```bash
model_server/example_client# python grpc_binary_client.py --grpc_port 9001 --input_name 0 --output_name 1463 --model_name resnet --batchsize 1
Start processing:
        Model name: resnet
        Images list file: input_images.txt
Traceback (most recent call last):
  File "grpc_binary_client.py", line 71, in <module>
    result = stub.Predict(request, 10.0) # result includes a dictionary with all model outputs
  File "/root/anaconda3/envs/openvino_client/lib/python3.7/site-packages/grpc/_channel.py", line 946, in __call__
    return _end_unary_response_blocking(state, call, False, None)
  File "/root/anaconda3/envs/openvino_client/lib/python3.7/site-packages/grpc/_channel.py", line 849, in _end_unary_response_blocking
    raise _InactiveRpcError(state)
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.INVALID_ARGUMENT
        details = "Received binary image input but resource not configured to accept NHWC layout"
        debug_error_string = "{"created":"@1631004339.947760100","description":"Error received from peer ipv6:[::1]:9001","file":"src/core/lib/surface/call.cc","file_line":1069,"grpc_message":"Received binary image input but resource not configured to accept NHWC layout","grpc_status":3}"
>
```

what should I do?

**Comment by mzegla** [comment]
Are you running OVMS with the same config as in the example - https://github.com/openvinotoolkit/model_server/blob/develop/src/custom_nodes/image_transformation/config_with_preprocessing_node.json

Looking at the message you got it seems that your model is not configured to accept input in NHWC layout, which is a requirement for using binary inputs feature. 
https://github.com/openvinotoolkit/model_server/blob/develop/src/custom_nodes/image_transformation/config_with_preprocessing_node.json#L7

Make sure you set `layout` parameter in your model configuration to `NHWC`.

**Comment by owlet42** [comment]
Yes, I used the configuration of the example, I did not set the default configuration.
The option 1 run success.
`libcustom_node_image_transformation.so` made by `make BASE_OS=centos`.

**Comment by mzegla** [comment]
Could you run model server in debug mode (use `--log_level DEBUG` with the command you launch model server with) and post logs here?

**Comment by owlet42** [comment]
I ran it again, this is the log.
<details>
<summary>log</summary>
```bash
[2021-09-09 02:59:44.131][1][serving][info][server.cpp:105] OpenVINO Model Server 2021.4.b0af59c9

[2021-09-09 02:59:44.151][1][serving][info][server.cpp:106] OpenVINO backend 2021.4.582

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:107] CLI parameters passed to ovms server

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:122] config_path: /workspace/config_with_preprocessing_node.json

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:124] gRPC port: 9001

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:125] REST port: 0

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:126] gRPC bind address: 0.0.0.0

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:127] REST bind address: 0.0.0.0

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:128] REST workers: 32

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:129] gRPC workers: 1

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:130] gRPC channel arguments: 

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:131] log level: DEBUG

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:132] log path: 

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:133] file system poll wait seconds: 1

[2021-09-09 02:59:44.151][1][serving][debug][server.cpp:134] sequence cleaner poll wait minutes: 5

[2021-09-09 02:59:44.152][1][modelmanager][debug][modelmanager.cpp:540] Loading configuration from /workspace/config_with_preprocessing_node.json

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:562] Specified model parameters:

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:563] model_basepath: /workspace/models/resnet50-caffe2-v1

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:564] model_name: resnet_model

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:565] batch_size: 0

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:567] shape: 1 3 224 224 

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:575] model_version_policy: latest: 1

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:577] nireq: 0

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:578] target_device: CPU

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:579] plugin_config:

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:581]   CPU_THROUGHPUT_STREAMS: 1

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:587] Batch size set: false, shape set: true

[2021-09-09 02:59:44.154][1][serving][debug][modelconfig.cpp:594] stateful: false

[2021-09-09 02:59:44.154][1][modelmanager][debug][modelmanager.cpp:903] Started applying config changes to model: resnet_model

[2021-09-09 02:59:44.158][1][modelmanager][debug][modelmanager.cpp:845] Detected version folder: 1

[2021-09-09 02:59:44.158][1][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 0

[2021-09-09 02:59:44.158][1][serving][info][model.cpp:38] Getting model from /workspace/models/resnet50-caffe2-v1

[2021-09-09 02:59:44.158][1][serving][info][model.cpp:45] Model downloaded to /workspace/models/resnet50-caffe2-v1

[2021-09-09 02:59:44.158][1][serving][info][model.cpp:145] Will add model: resnet_model; version: 1 ...

[2021-09-09 02:59:44.158][1][serving][debug][modelconfig.cpp:374] Parsing model: resnet_model mapping from path: /workspace/models/resnet50-caffe2-v1/1

[2021-09-09 02:59:44.160][1][serving][debug][model.cpp:119] Creating new model instance - model name: resnet_model; model version: 1;

[2021-09-09 02:59:44.160][1][serving][info][modelinstance.cpp:526] Loading model: resnet_model, version: 1, from path: /workspace/models/resnet50-caffe2-v1/1, with target device: CPU ...

[2021-09-09 02:59:44.160][1][serving][info][modelversionstatus.hpp:155] STATUS CHANGE: Version 1 of model resnet_model status change. New status: ( "state": "START", "error_code": "OK" )

[2021-09-09 02:59:44.160][1][serving][debug][modelversionstatus.hpp:122] setLoading: resnet_model - 1 (previous state: START) -> error: OK

[2021-09-09 02:59:44.160][1][serving][info][modelversionstatus.hpp:155] STATUS CHANGE: Version 1 of model resnet_model status change. New status: ( "state": "LOADING", "error_code": "OK" )

[2021-09-09 02:59:44.160][1][serving][debug][modelinstance.cpp:412] Getting model files from path: /workspace/models/resnet50-caffe2-v1/1

[2021-09-09 02:59:44.191][1][serving][debug][modelinstance.cpp:303] Try reading model file: /workspace/models/resnet50-caffe2-v1/1/resnet50-caffe2-v1-9.onnx

[2021-09-09 02:59:45.655][1][serving][debug][modelinstance.cpp:113] Network shape for input: gpu_0/data_0 - (1,3,224,224); final shape (1,3,224,224)

[2021-09-09 02:59:45.655][1][serving][debug][modelinstance.cpp:133] model: resnet_model, version: 1; reshaping inputs is not required

[2021-09-09 02:59:45.655][1][serving][info][modelinstance.cpp:160] Final network inputs: 

Input name: gpu_0/data_0; mapping_name: ; shape: (1,3,224,224); effective shape: (1,224,224,3); precision: FP32; layout: NHWC

[2021-09-09 02:59:45.655][1][serving][info][modelinstance.cpp:195] Output name: gpu_0/softmax_1; mapping name: ; shape: 1 1000 ; effective shape 1 1000 ; precision: FP32; layout: NC

[2021-09-09 02:59:46.244][1][serving][info][modelinstance.cpp:397] Plugin config for device CPU:

[2021-09-09 02:59:46.244][1][serving][info][modelinstance.cpp:401] CPU_THROUGHPUT_STREAMS: 1

[2021-09-09 02:59:46.244][1][serving][info][modelinstance.cpp:456] Loaded model resnet_model; version: 1; batch size: 1; No of InferRequests: 1

[2021-09-09 02:59:46.244][1][serving][debug][modelversionstatus.hpp:129] setAvailable: resnet_model - 1 (previous state: LOADING) -> error: OK

[2021-09-09 02:59:46.244][1][serving][info][modelversionstatus.hpp:155] STATUS CHANGE: Version 1 of model resnet_model status change. New status: ( "state": "AVAILABLE", "error_code": "OK" )

[2021-09-09 02:59:46.244][1][serving][info][model.cpp:85] Updating default version for model: resnet_model, from: 0

[2021-09-09 02:59:46.244][1][serving][info][model.cpp:95] Updated default version for model: resnet_model, to: 1

[2021-09-09 02:59:46.244][1][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:46.244][1][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:46.244][1][modelmanager][info][custom_node_library_manager.cpp:40] Loading custom node library name: image_transformation; base_path: /workspace/lib/libcustom_node_image_transformation.so

[2021-09-09 02:59:46.259][1][modelmanager][info][custom_node_library_manager.cpp:88] Successfully loaded custom node library name: image_transformation; base_path: /workspace/lib/libcustom_node_image_transformation.so

[2021-09-09 02:59:46.259][1][modelmanager][info][modelmanager.cpp:241] Reading pipeline: resnet configuration

[2021-09-09 02:59:46.259][1][modelmanager][debug][modelmanager.cpp:251] Pipeline: resnet does not have demultiply at entry node

[2021-09-09 02:59:46.259][1][serving][debug][modelmanager.cpp:191] Mapping node:request output:0, under alias:0

[2021-09-09 02:59:46.259][1][serving][debug][modelmanager.cpp:204] Mapping node: image_transformation_node model_name:  output: image, under alias: transformed_image

[2021-09-09 02:59:46.259][1][modelmanager][debug][modelmanager.cpp:306] Creating node: image_transformation_node type: custom model_name:  modelVersion: 0

[2021-09-09 02:59:46.259][1][serving][debug][modelmanager.cpp:170] Creating node dependencies mapping request. Node: image_transformation_node input: image <- SourceNode: request output: 0

[2021-09-09 02:59:46.259][1][serving][debug][modelmanager.cpp:204] Mapping node: resnet_node model_name: resnet_model output: gpu_0/softmax_1, under alias: 1463

[2021-09-09 02:59:46.259][1][modelmanager][debug][modelmanager.cpp:306] Creating node: resnet_node type: DL model model_name: resnet_model modelVersion: 0

[2021-09-09 02:59:46.259][1][serving][debug][modelmanager.cpp:170] Creating node dependencies mapping request. Node: resnet_node input: gpu_0/data_0 <- SourceNode: image_transformation_node output: transformed_image

[2021-09-09 02:59:46.259][1][serving][debug][modelmanager.cpp:170] Creating node dependencies mapping request. Node: response input: 1463 <- SourceNode: resnet_node output: 1463

[2021-09-09 02:59:46.259][1][serving][debug][modelmanager.cpp:329] Pipeline:resnet was not loaded so far. Triggering load

[2021-09-09 02:59:46.259][1][serving][info][modelchangesubscription.cpp:25] Subscription to model: resnet_model from resnet

[2021-09-09 02:59:46.259][1][modelmanager][debug][pipelinedefinition.cpp:49] Started validation of pipeline: resnet

[2021-09-09 02:59:46.259][1][modelmanager][debug][pipelinedefinition.cpp:933] Validation of pipeline definition: resnet nodes started.

[2021-09-09 02:59:46.259][1][modelmanager][debug][pipelinedefinition.cpp:311] Validation of pipeline: resnet; node name: request; node kind: 0

[2021-09-09 02:59:46.259][1][modelmanager][debug][pipelinedefinition.cpp:311] Validation of pipeline: resnet; node name: image_transformation_node; node kind: 2

[2021-09-09 02:59:46.260][1][modelmanager][debug][pipelinedefinition.cpp:311] Validation of pipeline: resnet; node name: resnet_node; node kind: 1

[2021-09-09 02:59:46.260][1][serving][debug][modelmanager.cpp:1030] Requesting model: resnet_model; version: 0.

[2021-09-09 02:59:46.260][1][serving][debug][model.hpp:78] Getting default version for model: resnet_model, 1

[2021-09-09 02:59:46.260][1][serving][debug][modelinstance.cpp:640] Model: resnet_model, version: 1 already loaded

[2021-09-09 02:59:46.260][1][modelmanager][debug][pipelinedefinition.cpp:311] Validation of pipeline: resnet; node name: response; node kind: 3

[2021-09-09 02:59:46.260][1][serving][debug][modelmanager.cpp:1030] Requesting model: resnet_model; version: 0.

[2021-09-09 02:59:46.260][1][serving][debug][model.hpp:78] Getting default version for model: resnet_model, 1

[2021-09-09 02:59:46.260][1][serving][debug][modelinstance.cpp:640] Model: resnet_model, version: 1 already loaded

[2021-09-09 02:59:46.260][1][serving][debug][model.hpp:78] Getting default version for model: resnet_model, 1

[2021-09-09 02:59:46.260][1][serving][debug][modelinstance.cpp:640] Model: resnet_model, version: 1 already loaded

[2021-09-09 02:59:46.260][1][modelmanager][debug][pipelinedefinition.cpp:82] Finished validation of pipeline: resnet

[2021-09-09 02:59:46.260][1][modelmanager][info][pipelinedefinition.cpp:83] Pipeline: resnet inputs: 

name: 0; shape: (1,0,0,0); effective shape: (1,0,0,0); precision: FP32; layout: ANY

[2021-09-09 02:59:46.261][1][modelmanager][info][pipelinedefinition.cpp:84] Pipeline: resnet outputs: 

name: 1463; shape: (1,1000); effective shape: (1,1000); precision: FP32; layout: NC

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinitionstatus.hpp:68] Pipeline: resnet state: BEGIN handling: ValidationPassedEvent: 

[2021-09-09 02:59:46.261][1][modelmanager][info][pipelinedefinitionstatus.hpp:76] Pipeline: resnet state changed to: AVAILABLE after handling: ValidationPassedEvent: 

[2021-09-09 02:59:46.261][1][modelmanager][info][pipeline_factory.cpp:69] Loading pipeline definition: resnet succeeded

[2021-09-09 02:59:46.261][1][serving][debug][modelmanager.cpp:336] Pipeline:resnet is already loaded. Triggering reload

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinitionstatus.hpp:68] Pipeline: resnet state: AVAILABLE handling: ReloadEvent: 

[2021-09-09 02:59:46.261][1][modelmanager][info][pipelinedefinitionstatus.hpp:76] Pipeline: resnet state changed to: RELOADING after handling: ReloadEvent: 

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinition.cpp:236] Unsubscribing pipeline: resnet from model: resnet_model

[2021-09-09 02:59:46.261][1][serving][info][modelchangesubscription.cpp:37] Subscription to model: resnet_model from resnet removed

[2021-09-09 02:59:46.261][1][serving][info][modelchangesubscription.cpp:25] Subscription to model: resnet_model from resnet

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinition.cpp:49] Started validation of pipeline: resnet

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinition.cpp:933] Validation of pipeline definition: resnet nodes started.

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinition.cpp:311] Validation of pipeline: resnet; node name: request; node kind: 0

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinition.cpp:311] Validation of pipeline: resnet; node name: image_transformation_node; node kind: 2

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinition.cpp:311] Validation of pipeline: resnet; node name: resnet_node; node kind: 1

[2021-09-09 02:59:46.261][1][serving][debug][modelmanager.cpp:1030] Requesting model: resnet_model; version: 0.

[2021-09-09 02:59:46.261][1][serving][debug][model.hpp:78] Getting default version for model: resnet_model, 1

[2021-09-09 02:59:46.261][1][serving][debug][modelinstance.cpp:640] Model: resnet_model, version: 1 already loaded

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinition.cpp:311] Validation of pipeline: resnet; node name: response; node kind: 3

[2021-09-09 02:59:46.261][1][serving][debug][modelmanager.cpp:1030] Requesting model: resnet_model; version: 0.

[2021-09-09 02:59:46.261][1][serving][debug][model.hpp:78] Getting default version for model: resnet_model, 1

[2021-09-09 02:59:46.261][1][serving][debug][modelinstance.cpp:640] Model: resnet_model, version: 1 already loaded

[2021-09-09 02:59:46.261][1][serving][debug][model.hpp:78] Getting default version for model: resnet_model, 1

[2021-09-09 02:59:46.261][1][serving][debug][modelinstance.cpp:640] Model: resnet_model, version: 1 already loaded

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinition.cpp:82] Finished validation of pipeline: resnet

[2021-09-09 02:59:46.261][1][modelmanager][info][pipelinedefinition.cpp:83] Pipeline: resnet inputs: 

name: 0; shape: (1,0,0,0); effective shape: (1,0,0,0); precision: FP32; layout: ANY

[2021-09-09 02:59:46.261][1][modelmanager][info][pipelinedefinition.cpp:84] Pipeline: resnet outputs: 

name: 1463; shape: (1,1000); effective shape: (1,1000); precision: FP32; layout: NC

[2021-09-09 02:59:46.261][1][modelmanager][debug][pipelinedefinitionstatus.hpp:68] Pipeline: resnet state: RELOADING handling: ValidationPassedEvent: 

[2021-09-09 02:59:46.261][1][modelmanager][info][pipelinedefinitionstatus.hpp:76] Pipeline: resnet state changed to: AVAILABLE after handling: ValidationPassedEvent: 

[2021-09-09 02:59:46.262][52][modelmanager][info][modelmanager.cpp:662] Started model manager thread

[2021-09-09 02:59:46.262][1][serving][debug][server.cpp:214] Starting grpc servers: 1

[2021-09-09 02:59:46.262][53][modelmanager][info][global_sequences_viewer.cpp:76] Started sequence cleaner thread

[2021-09-09 02:59:46.263][1][serving][info][server.cpp:226] Server started on port 9001

[2021-09-09 02:59:47.262][52][modelmanager][debug][modelmanager.cpp:665] Models configuration and filesystem check cycle begin

[2021-09-09 02:59:47.263][52][modelmanager][debug][modelmanager.cpp:612] Checking if something changed with model versions

[2021-09-09 02:59:47.263][52][modelmanager][debug][modelmanager.cpp:903] Started applying config changes to model: resnet_model

[2021-09-09 02:59:47.266][52][modelmanager][debug][modelmanager.cpp:845] Detected version folder: 1

[2021-09-09 02:59:47.266][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:47.266][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:47.266][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:47.266][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:47.266][52][modelmanager][debug][modelmanager.cpp:674] Models configuration and filesystem check cycle end

[2021-09-09 02:59:48.266][52][modelmanager][debug][modelmanager.cpp:665] Models configuration and filesystem check cycle begin

[2021-09-09 02:59:48.267][52][modelmanager][debug][modelmanager.cpp:612] Checking if something changed with model versions

[2021-09-09 02:59:48.267][52][modelmanager][debug][modelmanager.cpp:903] Started applying config changes to model: resnet_model

[2021-09-09 02:59:48.269][52][modelmanager][debug][modelmanager.cpp:845] Detected version folder: 1

[2021-09-09 02:59:48.269][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:48.269][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:48.269][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:48.269][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:48.269][52][modelmanager][debug][modelmanager.cpp:674] Models configuration and filesystem check cycle end

[2021-09-09 02:59:49.269][52][modelmanager][debug][modelmanager.cpp:665] Models configuration and filesystem check cycle begin

[2021-09-09 02:59:49.270][52][modelmanager][debug][modelmanager.cpp:612] Checking if something changed with model versions

[2021-09-09 02:59:49.270][52][modelmanager][debug][modelmanager.cpp:903] Started applying config changes to model: resnet_model

[2021-09-09 02:59:49.273][52][modelmanager][debug][modelmanager.cpp:845] Detected version folder: 1

[2021-09-09 02:59:49.273][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:49.273][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:49.273][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:49.273][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:49.273][52][modelmanager][debug][modelmanager.cpp:674] Models configuration and filesystem check cycle end

[2021-09-09 02:59:50.273][52][modelmanager][debug][modelmanager.cpp:665] Models configuration and filesystem check cycle begin

[2021-09-09 02:59:50.274][52][modelmanager][debug][modelmanager.cpp:612] Checking if something changed with model versions

[2021-09-09 02:59:50.274][52][modelmanager][debug][modelmanager.cpp:903] Started applying config changes to model: resnet_model

[2021-09-09 02:59:50.277][52][modelmanager][debug][modelmanager.cpp:845] Detected version folder: 1

[2021-09-09 02:59:50.277][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:50.277][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:50.277][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:50.277][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:50.277][52][modelmanager][debug][modelmanager.cpp:674] Models configuration and filesystem check cycle end

[2021-09-09 02:59:50.483][57][serving][debug][prediction_service.cpp:74] Processing gRPC request for model: resnet; version: 0

[2021-09-09 02:59:50.483][57][serving][debug][modelmanager.cpp:1030] Requesting model: resnet; version: 0.

[2021-09-09 02:59:50.483][57][serving][debug][prediction_service.cpp:83] Requested model: resnet does not exist. Searching for pipeline with that name...

[2021-09-09 02:59:50.483][57][serving][debug][modelmanager.cpp:1055] Requesting pipeline: resnet;

[2021-09-09 02:59:50.483][57][serving][debug][pipelinedefinition.cpp:123] Successfully waited for pipeline definition: resnet

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipelinedefinition.cpp:175] Creating pipeline: resnet. Adding nodeName: request, modelName: 

[2021-09-09 02:59:50.483][57][dag_executor][debug][node.cpp:52] Will create node: request with demultiply: NA, gatherFrom: NA.

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipelinedefinition.cpp:175] Creating pipeline: resnet. Adding nodeName: image_transformation_node, modelName: 

[2021-09-09 02:59:50.483][57][dag_executor][debug][node.cpp:52] Will create node: image_transformation_node with demultiply: NA, gatherFrom: NA.

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipelinedefinition.cpp:175] Creating pipeline: resnet. Adding nodeName: resnet_node, modelName: resnet_model

[2021-09-09 02:59:50.483][57][dag_executor][debug][node.cpp:52] Will create node: resnet_node with demultiply: NA, gatherFrom: NA.

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipelinedefinition.cpp:175] Creating pipeline: resnet. Adding nodeName: response, modelName: 

[2021-09-09 02:59:50.483][57][dag_executor][debug][node.cpp:52] Will create node: response with demultiply: NA, gatherFrom: NA.

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipelinedefinition.cpp:217] Connecting pipeline: resnet, from: resnet_node, to: response

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipeline.cpp:42] Connecting from: resnet_node, to: response

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipeline.cpp:57] Links from:resnet_node to:response:

response[1463]=resnet_node[1463]


[2021-09-09 02:59:50.483][57][dag_executor][debug][pipelinedefinition.cpp:217] Connecting pipeline: resnet, from: image_transformation_node, to: resnet_node

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipeline.cpp:42] Connecting from: image_transformation_node, to: resnet_node

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipeline.cpp:57] Links from:image_transformation_node to:resnet_node:

resnet_node[gpu_0/data_0]=image_transformation_node[transformed_image]


[2021-09-09 02:59:50.483][57][dag_executor][debug][pipelinedefinition.cpp:217] Connecting pipeline: resnet, from: request, to: image_transformation_node

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipeline.cpp:42] Connecting from: request, to: image_transformation_node

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipeline.cpp:57] Links from:request to:image_transformation_node:

image_transformation_node[image]=request[0]


[2021-09-09 02:59:50.483][57][dag_executor][debug][pipeline.cpp:83] Started execution of pipeline: resnet

[2021-09-09 02:59:50.483][57][dag_executor][debug][node.cpp:164] Will create new session:  for node: request

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipeline.cpp:111] Pipeline: resnet got message that node: request session:  finished.

[2021-09-09 02:59:50.483][57][dag_executor][debug][pipeline.cpp:119] Fetching results of pipeline: resnet node: request session: 

[2021-09-09 02:59:50.483][57][dag_executor][debug][entry_node.cpp:82] [Node: request] Deserializing input: 0

[2021-09-09 02:59:50.483][57][dag_executor][debug][entry_node.cpp:97] Request contains binary input: image

[2021-09-09 02:59:50.483][57][dag_executor][debug][node.cpp:68] Will remove node: request session: 

[2021-09-09 02:59:50.483][57][dag_executor][warning][pipeline.cpp:121] Executing pipeline: resnet node: request session:  failed with ret code: 180, error message: Received binary image input but resource not configured to accept NHWC layout

[2021-09-09 02:59:51.277][52][modelmanager][debug][modelmanager.cpp:665] Models configuration and filesystem check cycle begin

[2021-09-09 02:59:51.278][52][modelmanager][debug][modelmanager.cpp:612] Checking if something changed with model versions

[2021-09-09 02:59:51.279][52][modelmanager][debug][modelmanager.cpp:903] Started applying config changes to model: resnet_model

[2021-09-09 02:59:51.282][52][modelmanager][debug][modelmanager.cpp:845] Detected version folder: 1

[2021-09-09 02:59:51.282][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:51.282][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:51.282][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:51.282][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:51.282][52][modelmanager][debug][modelmanager.cpp:674] Models configuration and filesystem check cycle end

[2021-09-09 02:59:52.282][52][modelmanager][debug][modelmanager.cpp:665] Models configuration and filesystem check cycle begin

[2021-09-09 02:59:52.283][52][modelmanager][debug][modelmanager.cpp:612] Checking if something changed with model versions

[2021-09-09 02:59:52.283][52][modelmanager][debug][modelmanager.cpp:903] Started applying config changes to model: resnet_model

[2021-09-09 02:59:52.286][52][modelmanager][debug][modelmanager.cpp:845] Detected version folder: 1

[2021-09-09 02:59:52.286][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:52.286][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:52.286][52][modelmanager][debug][modelmanager.cpp:701] Currently registered model: resnet_model versions count: 1

[2021-09-09 02:59:52.286][52][modelmanager][debug][modelmanager.cpp:703] model: resnet_model version: 1 state: AVAILABLE

[2021-09-09 02:59:52.286][52][modelmanager][debug][modelmanager.cpp:674] Models configuration and filesystem check cycle end
```
</details>

**Comment by mzegla** [comment]
I noticed that you were using contents from `develop` branch and model server image in version 2021.4 
The example you follow requires changes in OVMS that were introduced in 2021.4.1. 
We recently released 2021.4.1, so latest docker image supports that use case and the stuff you used on `develop` branch is now merged to `main`. Please, update your docker image to 2021.4.1 and try again.

**Comment by owlet42** [comment]
It's working, think you!
